src/modules/recipes/api.ts

```diff
@@ -0,0 +1,46 @@
+import { Recipe, ApiResponse } from "./models";
+import { fetchData } from "../../utils";
+
+const API_BASE_URL = "https://www.themealdb.com/api/json/v1/1";
+
+// fetchRecipeById fonksiyonu
+export const fetchRecipeById = async (id: string): Promise<Recipe> => {
+  const fullUrl = `${API_BASE_URL}/lookup.php?i=${id}`;
+  const data: ApiResponse<Recipe> = await fetchData(fullUrl);
+
+  if (data.meals && data.meals.length > 0) {
+    const recipe = data.meals[0];
+    const ingredients: string[] = [];
+    for (let i = 1; i <= 20; i++) {
```

Comment:
I would not recommend modifying the data that you fetch from API here. This function should just fetch the data in whatever format the data is.
Later, the data will be stored in the Context in the same format.

Whenever you want to display data in a modified way, then you'll need to add this modification logic there.